daemon/main.go

```diff
@@ -127,6 +132,22 @@ func main() {
 	}
 }
 
+func setupPidFile() {
+	pidfile.SetPidfilePath(defaults.PidFilePath)
+	if err := pidfile.Write(); err != nil {
+		log.WithError(err).Fatal("could not create pidfile")
+	}
+
+	// Handle the cleanup
+	sig := make(chan os.Signal, 1)
```

Comment:
I think you should move this to a new function called clean up. We might use it to do more cleanups besides `os.Remove(defaults.PidFilePath)`

Reply:
fixed